app/policies/department_policy.rb

```diff
@@ -20,6 +20,29 @@ def initialize(user, department)
     @department = department
   end
 
+  def permitted_attributes
+    return [user_ids: [], contact_attributes: contact_attrs] if @user.superadmin?
+    return [contact_attributes: contact_attrs] if @user.department_manager?
+    []
+  end
+
+  def phone_attrs
+    [:id, :body, :label, :_destroy, :type, :contacts_id]
+  end
+
+  def email_attrs
+    [:id, :body, :label, :_destroy, :type, :contacts_id]
+  end
+
+  def contact_attrs
+    [
+      :id, :name, :_destroy, :contactable_id, :contactable_type, :street,
+      :extended, :city, :postal_code,
+      contact_emails_attributes: email_attrs,
+      contact_phones_attributes: phone_attrs
+    ]
+  end
+
```

Comment:
@kitsane can we do this week a coaching session with the topic related to these lines 23-44:
1. Action Controller Parameters
2. technical question: why is it better to move these lines in the policy instead in a concern? or what is the difference here?